src/dodal/devices/areadetector/andor2_epics/andor2_controller.py

```diff
@@ -0,0 +1,79 @@
+import asyncio
+
+from ophyd_async.core import DetectorControl, DetectorTrigger
+from ophyd_async.core._detector import TriggerInfo
+from ophyd_async.epics import adcore
+from ophyd_async.epics.adcore import (
+    DEFAULT_GOOD_STATES,
+    DetectorState,
+    stop_busy_record,
+)
+
+from .andor2_io import (
+    Andor2DriverIO,
+    Andor2TriggerMode,
+    ImageMode,
+)
+
+
+class Andor2Controller(DetectorControl):
+    """
+    Andor 2 controller
+
+    """
+
+    _supported_trigger_types = {
+        DetectorTrigger.internal: Andor2TriggerMode.internal,
+        DetectorTrigger.constant_gate: Andor2TriggerMode.ext_trigger,
+    }
+
+    def __init__(
+        self,
+        driver: Andor2DriverIO,
+        good_states: set[DetectorState] | None = None,
+    ) -> None:
+        if good_states is None:
+            good_states = set(DEFAULT_GOOD_STATES)
+        self._drv = driver
+        self.good_states = good_states
+
+    def get_deadtime(self, exposure: float | None) -> float:
+        if exposure is None:
+            return 0.1
+        return exposure + 0.1
```

Comment:
Nit: I would pull the 0.1 out into a constant of `DEAD_TIME` or something, aids in readability + maintainability.